internal/saucecloud/cloud.go

```diff
@@ -985,12 +987,27 @@ func (r *CloudRunner) getHistory(launchOrder config.LaunchOrder) (insights.JobHi
 	return r.InsightsService.GetHistory(context.Background(), user, launchOrder)
 }
 
+func getSource(isRDC bool) build.Source {
+	if isRDC {
+		return build.RDC
+	}
+	return build.VDC
+}
+
 func (r *CloudRunner) reportSuiteToInsights(res result) {
 	// Skip reporting if job is not completed
 	if !job.Done(res.job.Status) || res.skipped || res.job.ID == "" {
 		return
 	}
 
+	if res.details.BuildID == "" {
+		buildID, err := r.BuildService.GetBuildID(context.Background(), res.job.ID, getSource(res.job.IsRDC))
+		if err != nil {
+			log.Warn().Err(err).Str("action", "getBuild").Str("jobID", res.job.ID).Msg(msg.InsightsReportError)
+			return
```

Comment:
So if we can't get a build ID, we won't report the results to Insights at all. Do we need to be this strict? 🤔 Or can we proceed to report, but simply without the build ID. Would there be some negative consequence because of it? In other words, what's worse: Not reporting at all, or a report with a missing build ID?

Reply:
I think an empty build ID is better. Let me modify it.